File: testers/pg/java/src/java/adventureworks/DomainInsertImpl.java
package adventureworks;

import adventureworks.public_.*;
import java.util.Random;
import typr.runtime.internal.RandomHelper;

public class DomainInsertImpl implements TestDomainInsert {
  @Override
  public Flag publicFlag(Random random) {
    return new Flag(random.nextBoolean());
  }

  @Override
  public Mydomain publicMydomain(Random random) {
    return new Mydomain(RandomHelper.alphanumeric(random, 10));
  }

  @Override
  public Name publicName(Random random) {
    return new Name(RandomHelper.alphanumeric(random, 10));
  }

  @Override
  public NameStyle publicNameStyle(Random random) {
    return new NameStyle(random.nextBoolean());
  }

  @Override
  public ShortText publicShortText(Random random) {
    return new ShortText(RandomHelper.alphanumeric(random, 10));
  }
}

File: testers/pg/scala/javatypes/src/scala/adventureworks/DomainInsertImpl.scala
package adventureworks

import adventureworks.public.*
import typr.runtime.internal.RandomHelper

import java.util.Random

class DomainInsertImpl extends TestDomainInsert {
  override def publicFlag(random: Random): Flag =
    Flag(random.nextBoolean())

  override def publicMydomain(random: Random): Mydomain =
    Mydomain(RandomHelper.alphanumeric(random, 10))

  override def publicName(random: Random): Name =
    Name(RandomHelper.alphanumeric(random, 10))

  override def publicNameStyle(random: Random): NameStyle =
    NameStyle(random.nextBoolean())

  override def publicShortText(random: Random): ShortText =
    ShortText(RandomHelper.alphanumeric(random, 10))
}
